lib/src/values/big_logic_value.dart

```diff
@@ -10,6 +10,25 @@
 
 part of values;
 
+/// Extends [BigInt] with utility functions that are useful for dealing with
+/// large bit vectors and conversion between types.
+extension BigLogicValueBigIntUtilities on BigInt {
+  /// Returns this [BigInt] as an [int].
+  ///
+  /// Always interprets the number as unsigned, and thus never clamps to fit.
+  int toIntUnsigned(int width) {
+    if (width > LogicValue._INT_BITS) {
+      throw Exception('Cannot convert to BigInt when width $width'
+          ' is greater than ${LogicValue._INT_BITS}');
+    } else if (width == LogicValue._INT_BITS) {
+      // With `0x` in front of a hex literal, will be interpreted as unsigned.
+      return int.parse('0x${toRadixString(16)}');
```

Comment:
There's got to be a better way to get at this; maybe 32 bits at a time can be safely extracted from the bigint through bit shifting?

Reply:
Sorry, looks like I just barely merged before your comment!

You're right, this appears to work well also, and is probably better?  What do you think?
```dart
  int toIntUnsigned(int width) {
    if (width > LogicValue._INT_BITS) {
      throw Exception('Cannot convert to BigInt when width $width'
          ' is greater than ${LogicValue._INT_BITS}');
    } else if (width == LogicValue._INT_BITS) {
      const maskWidth = 32;
      final mask = _BigLogicValue._maskOfWidth(maskWidth);
      return (this & mask).toInt() |
          (((this >> maskWidth) & mask).toInt() << maskWidth);
    } else {
      return toInt();
    }
  }
```

Reply:
Thanks for noticing my late comment.  I see no correctness issues in this code.  Unfortunately it's a bit more complex than I expected, but I don't see a nice way to clean that up.  Despite version control history recording why these changes were made, I still suggest adding a comment explaining that this intentionally complex code is done this way to avoid signed conversion.  
Finally, I worry a bit about what happens when this all is run in a javascript environment where _INT_BITS isn't 64.  If _INT_BITS isn't even, even adjusting maskWidth won't be able to fix this code.  But maybe that's a problem for the future.

Reply:
Good point, I'll add a comment.

There's already a lot of problems with javascript that I'm noticing.  For example, in this PR I disabled the lint checks for large constants already (like `0xffffffffffffffff`) that won't work with javascript.  I think there's going to be a number of surprises when compiling to js, unfortunately.